SAGE Journals.js

```diff
@@ -116,36 +117,35 @@ function scrape(doc, url) {
 					item.title += ': ' + subtitle.trim();
 				}
 			}
-			// The encoding of apostrophs in the RIS are incorrect and
-			// therefore we extract the abstract again from the website.
+			
+			// ubtue: extract translated and other abstracts from the different xpath
+			var ubtueabstract = ZU.xpathText(doc, '//article//div[contains(@class, "abstractSection")]/p');
+			var otherabstract = ZU.xpathText(doc, '//article//div[contains(@class, "tabs-translated-abstract")]/p');
 			var abstract = ZU.xpathText(doc, '//article//div[contains(@class, "abstractSection")]/p');
-			if (abstract) {
-				item.abstractNote = abstract.trim();
-			}
-
-			// ubtue: also add translated abstracts in 520
-			var translatedAbstract = ZU.xpathText(doc, '//article//div[contains(@class, "tabs-translated-abstract")]/p');
-			if (translatedAbstract) {
+			if (ubtueabstract && otherabstract) {
+				item.abstractNote = ubtueabstract;
 				item.notes.push({
-					note: "abs:" + translatedAbstract,
+					note: "abs:" + otherabstract.replace(/^Résumé/, ''),
 				});
+			} else if (ubtueabstract && !otherabstract) {
+				ubtueabstract = ZU.xpathText(doc, '//*[contains(concat( " ", @class, " " ), concat( " ", "abstractInFull", " " ))]');
+				item.abstractNote = ubtueabstract;
+			} else if (!ubtueabstract && !otherabstract) {
+				ubtueabstract = ZU.xpathText(doc, '//*[contains(concat( " ", @class, " " ), concat( " ", "abstractInFull", " " ))]');
+				item.abstractNote = ubtueabstract;
+			} else {
+				item.abstractNote = abstract;
 			}
 
-			// Only get attributes and skip introductory term Keywords
-			let tagentry = ZU.xpathText(doc, '//kwd-group[1] | //*[contains(concat( " ", @class, " " ), concat( " ", "hlFld-KeywordText", " " ))] \
-																  /a[@class="attributes"]');
-			// Make sure we keep the original approach
-			if (!tagentry)
-				tagentry = ZU.xpathText(doc, '//kwd-group[1] | //*[contains(concat( " ", @class, " " ), concat( " ", "hlFld-KeywordText", " " ))]');
+			var tagentry = ZU.xpathText(doc, '//kwd-group[1] | //*[contains(concat( " ", @class, " " ), concat( " ", "hlFld-KeywordText", " " ))]');
 			if (tagentry) {
-				item.tags = tagentry.split(",");
+				item.tags = tagentry.replace(/.*Keywords/, ',').replace(/Mots-clés/, ',').split(",");
 			}
-
-			// ubtue: add tags "Book Review" if "Review Article"
+			// ubtue: add tags "Book Review" if "Product article" or "Book review"
 			if (articleType) {
 				for (let r of articleType) {
 					let reviewDOIlink = r.innerHTML;
-					if (reviewDOIlink.match(/Review Article/)) {
+					if (reviewDOIlink.match(/(product|book)\s+reviews?/i)) {
```

Comment:
What about the `Review Article` cases that originally matched?

Reply:
`Review Article` will pushed to keywords and will not be treated as review articles. see my recent changes